### DISCUSSION TITLE
ERROR: syntax error at or near "CREATE_REPLICATION_SLOT" (SQLSTATE 42601)

### DISCUSSION THREAD

**Discussion by Amogh-Bharadwaj** [cloudnative-pg/cloudnative-pg#4950] [Q&A]
Hi! 
Currently encountering the above error when I try to programmatically create a replication slot in Golang using [pglogrepl](https://github.com/jackc/pglogrepl), for a self-hosted Postgres primary database with CloudNativePG:
```golang
		opts := pglogrepl.CreateReplicationSlotOptions{
			Temporary: false,
			Mode:      pglogrepl.LogicalReplication,
		}
		res, err := pglogrepl.CreateReplicationSlot(ctx, conn.PgConn(), slot, "pgoutput", opts)
```
Postgres version: 16

Is logical replication supported on CloudNativePG? Is there some setting which I can set to enable the above operation?

**Comment by mnencia** [comment]
You are probably using a normal connection, but Postgres accepts that command only on a replication connection. See https://www.postgresql.org/docs/current/protocol-replication.html for more information about opening a replication connection.

**Reply by Amogh-Bharadwaj** [reply]
Hi @mnencia !
Thanks for the response 

Actually we do use a replication connection and this works with all other Postgres offerings
